tests/test_inference.py

```diff
@@ -50,18 +50,12 @@ def get_random_data_example(num_samples, num_sites, seed=42, num_states=2):
 
 
 class TestUnfinalisedErrors:
-    def make_ancestor_data_unfinalised(self, path=None):
-        with tsinfer.SampleData(path=path, sequence_length=2) as sample_data:
-            sample_data.add_site(1, genotypes=[0, 1, 1, 0], alleles=["G", "C"])
-            with pytest.raises(ValueError):
-                tsinfer.AncestorData(sample_data)
-        if path is not None:
-            sample_data.close()
-
     def match_ancestors_ancestors_unfinalised(self, path=None):
         with tsinfer.SampleData(sequence_length=2) as sample_data:
             sample_data.add_site(1, genotypes=[0, 1, 1, 0], alleles=["G", "C"])
-        with tsinfer.AncestorData(sample_data, path=path) as ancestor_data:
+        with tsinfer.AncestorData(
```

Comment:
Ditto, `sd` rather than `sample_data` might make it all more readable (fewer lines)